src/CodeStyle/Core/Analyzers/AbstractFormattingAnalyzer.cs

```diff
@@ -1,44 +1,51 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System.Collections.Immutable;
 using System.IO;
 using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Formatting;
 using Microsoft.CodeAnalysis.Options;
 using Microsoft.VisualStudio.CodingConventions;
 
 namespace Microsoft.CodeAnalysis.CodeStyle
 {
-    internal abstract class AbstractFormattingAnalyzerImpl
+    internal abstract class AbstractFormattingAnalyzer
+        : AbstractCodeStyleDiagnosticAnalyzer
     {
-        private readonly DiagnosticDescriptor _descriptor;
-
-        protected AbstractFormattingAnalyzerImpl(DiagnosticDescriptor descriptor)
+        protected AbstractFormattingAnalyzer()
+            : base(
+                IDEDiagnosticIds.FormattingDiagnosticId,
+                new LocalizableResourceString(nameof(CodeStyleResources.Fix_formatting), CodeStyleResources.ResourceManager, typeof(CodeStyleResources)),
+                new LocalizableResourceString(nameof(CodeStyleResources.Fix_formatting), CodeStyleResources.ResourceManager, typeof(CodeStyleResources)))
         {
-            _descriptor = descriptor;
         }
 
-        internal void InitializeWorker(AnalysisContext context)
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            => ImmutableArray.Create(Descriptor);
+
+        protected abstract ISyntaxFormattingService SyntaxFormattingService { get; }
+
+        protected override void InitializeWorker(AnalysisContext context)
         {
-            var workspace = new AdhocWorkspace();
             var codingConventionsManager = CodingConventionsManagerFactory.CreateCodingConventionsManager();
```

Comment:
>codingConventionsManager [](start = 16, length = 24)

Nit: Can this be a readonly field?

Reply:
➡️ It could be, but it doesn't really make a difference here. This will go away anyway once we move to the compiler .editorconfig APIs.